packages/effect/src/Config.ts

```diff
@@ -367,6 +367,14 @@ export const hashSet: <A>(config: Config<A>, name?: string) => Config<HashSet.Ha
  */
 export const string: (name?: string) => Config<string> = internal.string
 
+/**
+ * Constructs a config for a non-empty string value.
+ *
+ * @since 3.7.0
+ * @category constructors
+ */
+export const nonEmptyString: (name?: string) => Config<string> = internal.nonEmptyString
```

Comment:
I think it should be called `stringNonEmpty`

Reply:
note that there's already a `Match.nonEmptyString`

Reply:
Maybe we can consolidate at some point.

I prefer `stringNonEmpty` as the developer can start typing "string" and see all the options.

Reply:
> developer can start typing "string" and see all the options

yeah that's nice, however until we decide to make the change across the board, I think it would be better to maintain consistency

there are also `NonEmptyReadonlyArray`, `Schema.nonEmptyString`, `Schema.NonEmptyArray`, etc...

Reply:
Yeah sure, can put it on the 4.0 list.